src/App.tsx

```diff
@@ -1,26 +1,198 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { todos as todosClient } from './api/todos';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoHead } from './Components/TodoHead';
+import { TodoList } from './Components/TodoList';
+import { TodoFilter } from './Components/TodoFilter';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+  const [inputResetFlag, setInputResetFlag] = useState(Math.random());
+  const [inputLoadingFlag, setInputLoadingFlag] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<string | null>(null);
+
+  const handleErrors = (error: Errors) => {
+    window.setTimeout(() => {
+      setErrorMessage(error);
+    }, 300);
+  };
+
+  const filtering = (arr: Todo[], filterParam: Filters): Todo[] | [] => {
```

Comment:
```suggestion
  const filterTodos = (todos: Todo[], filterParam: Filters): Todo[] | [] => {
```
And define it outside of the component

arr is bad name